### PR TITLE
🐛 Remove overlay entry only when it's been inserted

### LINKED ISSUE #65: debug模式enable为false时重建导致app异常卡死
## Steps to Reproduce 复现步骤

1. debug模式未启动ume：
`final openUme = false.obs;`
`runApp(Obx(() {
          return UMEWidget(
            child: MyApp(),
            enable: openUme.value,
          );
        }));`
2. 使用`Get.updateLocale(Locale.fromSubtags(languageCode: val));`切换语言

## Error log 错误日志

``` bash

异常原因：'package:flutter/src/widgets/overlay.dart': Failed assertion: line 147 pos 12: '_overlay != null': is not true.
```

## 原因
未启动ume时，`_UMEWidgetState`的`OverlayEntry _overlayEntry = OverlayEntry(builder: (ctx) => Container());`
虽然有值，但是没有插入显示，所以`_overlayEntry`属性`OverlayState? _overlay;`为空的。
但是当重建时执行了`void _removeOverlay() => _overlayEntry.remove();`，
然后调用了`assert(_overlay != null);`这里debug模式断言失败导致app异常卡死。
我本地修改`_overlayEntry`为可空就没问题了。
``` bash
class _UMEWidgetState extends State<UMEWidget> {
//  OverlayEntry _overlayEntry = OverlayEntry(builder: (ctx) => Container());
OverlayEntry?  _overlayEntry ;


}

```

### PR DESCRIPTION
**Pull request by AlexV525**
Fix #65.

## Pull Request Checklist

- [x] I have read the [UME contribution document](../CONTRIBUTING.md) and understand how to contribute, commit the code according to the rules. 我已阅读过 UME 贡献文档，并了解如何进行贡献，按照规则提交了代码
- [x] I have added the necessary comments in code to ensure that other contributors can understand the reason for the change. 我在修改中已经添加了必要的注释，以确保便于其他贡献者理解修改原因
- [x] The code has been formatted by dartfmt before push. 代码在提交前已经经过 dartfmt 进行了格式化
- [x] Change does not involve the adjustment of test cases. Or all existing and new tests are passing. 改动不涉及测试用例调整，或 example 工程与单元测试已经完全跑通

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -20,72 +19,72 @@ final Dio dio = Dio()..options = BaseOptions(connectTimeout: 10000);
...
+  State<UMEApp> createState() => _UMEAppState();
 }
 
-class _MyAppState extends State<MyApp> {
+class _UMEAppState extends State<UMEApp> {
   @override
   void initState() {
     super.initState();
-    WidgetsBinding.instance?.addPostFrameCallback((_) {
-      CustomRouterPluggable().navKey = navigatorKey;
```
Comment (talisk) [inline example/lib/main.dart:67]: It's the global navigator key. `CustomRouterPluggable` need it and build the "Anywhere door".

Comment (talisk) [reply example/lib/main.dart:67]: We should keep the line. And find an appropriate time to set it.

**Review by talisk — CHANGES_REQUESTED** [review]

File: example/lib/main.dart (hunk 1 above)
Comment (AlexV525) [reply example/lib/main.dart:67]: Oh it's an accident. Will add it back later.

**Review by talisk — APPROVED** [review]
LGTM

**Review by talisk — APPROVED** [review]
LGTM

**Pull request merged**